subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -54,166 +54,54 @@ bool nrf_cloud_agps_request_in_progress(void)
 	return atomic_get(&request_in_progress) != 0;
 }
 
-#if IS_ENABLED(CONFIG_NRF_CLOUD_MQTT)
-static int json_add_types_array(cJSON * const obj, enum nrf_cloud_agps_type *types,
-				const size_t type_count)
-{
-	__ASSERT_NO_MSG(obj != NULL);
-	__ASSERT_NO_MSG(types != NULL);
-
-	cJSON *array;
-
-	if (!type_count) {
-		return -EINVAL;
-	}
-
-	array = cJSON_AddArrayToObject(obj, NRF_CLOUD_JSON_KEY_AGPS_TYPES);
-	if (!array) {
-		return -ENOMEM;
-	}
-
-	for (size_t i = 0; i < type_count; i++) {
-		cJSON_AddItemToArray(array, cJSON_CreateNumber(types[i]));
-	}
-
-	if (cJSON_GetArraySize(array) != type_count) {
-		cJSON_DeleteItemFromObject(obj, NRF_CLOUD_JSON_KEY_AGPS_TYPES);
-		return -ENOMEM;
-	}
-
-	return 0;
-}
-#endif /* IS_ENABLED(CONFIG_NRF_CLOUD_MQTT) */
-
 int nrf_cloud_agps_request(const struct nrf_modem_gnss_agps_data_frame *request)
 {
 #if IS_ENABLED(CONFIG_NRF_CLOUD_MQTT)
 	if (nfsm_get_current_state() != STATE_DC_CONNECTED) {
 		return -EACCES;
 	}
+	if (!request) {
+		return -EINVAL;
+	}
 
 	int err;
-	enum nrf_cloud_agps_type types[9];
-	size_t type_count = 0;
-	cJSON *data_obj;
 	cJSON *agps_req_obj;
-	uint32_t ephem;
+	struct nrf_modem_gnss_agps_data_frame req = *request;
```

Comment:
Maybe the function should just accept `request` as a struct-by-value rather than by reference? Might as well, since we're already copying it into the stack here

Reply:
Or perhaps you are doing it this way just for external consistency, which makes sense

In which case, I'd just add a comment pointing out that this line is copying the struct into stack, and not re-casting the pointer to non-const -- it's pretty easy to misread the line as the latter rather than the former.

`/* Copy the request struct into the stack so that we can modify it before passing it along to such and such */`